src/chain/chain-supervision.h

```diff
@@ -256,6 +269,19 @@ bool ProtoSupervisionToSupervision(
     const ProtoSupervision &proto_supervision,
     Supervision *supervision);
 
+/** This function creates and initializes an end-to-end supervision object
+    from a training FST (e.g. created using compile-train-graphs). It simply
+    sets all the input and output labels to pdf_id+1 (i.e. converts the FST to
+    an FSA) and stores the resulting FST in e2e-fsts[0].
+    It returns true if there were no epsilon transitions, otherwise
+    it would return false.
+    To find out more about end-to-end training, see chain-generic-numerator.h
+ */
+bool TrainingGraphToSupervision(
```

Comment:
I think you should add 'E2e' to the end of this function name to indicate that it relates to the e2e setup.